apps/condo/domains/news/components/NewsForm/BaseNewsForm.tsx

```diff
@@ -0,0 +1,853 @@
+import { NEWS_TYPE_COMMON, NEWS_TYPE_EMERGENCY } from '@condo/domains/news/constants/newsTypes'
+import {
+    NewsItemScope,
+    NewsItem,
+} from '@condo/domains/news/utils/clientSchema'
+import { Property } from '@condo/domains/property/utils/clientSchema'
+import { searchOrganizationProperty } from '@condo/domains/ticket/utils/clientSchema/search'
+import { SectionNameInput } from '@condo/domains/user/components/SectionNameInput'
+import { UnitNameInput, UnitNameInputOption } from '@condo/domains/user/components/UnitNameInput'
+
+type FormWithActionChildrenProps = ComponentProps<ComponentProps<typeof FormWithAction>['children']>
+
+type ActionBarProps = Pick<FormWithActionChildrenProps, 'handleSave' | 'isLoading' | 'form'>
+
+type ActionNameProps = 'create' | 'update'
+
+type NewsItemClientUtilsType = IGenerateHooksResult<INewsItem, INewsItemCreateInput, INewsItemUpdateInput, IQueryAllNewsItemsArgs>
+
+export type SendPeriodType = 'now' | 'later'
+
+export type BaseNewsFormProps = {
+    organizationId: string
+    ActionBar: React.FC<ActionBarProps>
+    action: (values: INewsItemCreateInput | INewsItemUpdateInput) => ReturnType<ReturnType<NewsItemClientUtilsType['useCreate' | 'useUpdate']>>
+    initialValues?: INewsItem
+    & {
+        newsItemScopes: INewsItemScope[],
+        hasAllProperties: boolean,
+        sendPeriod: SendPeriodType,
+    },
+    templates: { [key: string]: Pick<INewsItemTemplate, 'title' | 'body'> }
+    afterAction?: () => void,
+    newsItem?: INewsItem,
+    OnCompletedMsg: (INewsItem) => ArgsProps | null,
+    allNews: INewsItem[],
+    actionName: ActionNameProps,
+}
+
+const COUNTER_COL_STYLE: React.CSSProperties = { position: 'absolute', right: 0, bottom: 0, margin: '12px', padding: '2px 10px', backgroundColor: 'black', borderRadius: '100px' }
+const FORM_FILED_COL_PROPS = { style: { width: '100%', padding: 0 } }
+export const SCROLL_TO_FIRST_ERROR_CONFIG: ScrollOptions = { behavior: 'smooth', block: 'center' }
+export const SHOW_TIME_CONFIG = { defaultValue: dayjs('00:00:00:000', 'HH:mm:ss:SSS') }
+export const FULL_WIDTH_STYLE: React.CSSProperties = { width: '100%' }
+const BIG_VERTICAL_GUTTER: [Gutter, Gutter] = [0, 60]
+const MEDIUM_VERTICAL_GUTTER: [Gutter, Gutter] = [0, 40]
+const SMALL_VERTICAL_GUTTER: [Gutter, Gutter] = [0, 24]
+const BIG_HORIZONTAL_GUTTER: [Gutter, Gutter] = [50, 0]
+
+const getIsDateInFuture = (form, fieldName) => {
+    const date = form.getFieldsValue([fieldName])[fieldName]
+    return date.isSameOrAfter(dayjs(), 'minute')
+}
+const getValidBeforeAfterSendAt = (form) => {
+    const { sendAt, validBefore } = form.getFieldsValue(['sendAt', 'validBefore'])
+    if (sendAt && validBefore) {
+        const diff = dayjs(validBefore).diff(sendAt)
+        if (diff < 0) return false
+    }
+    return true
+}
+
+const containWordsInBrackets = (str) => {
+    const getAllSquareBracketsOccurrencesRegex  = /\[[^\]]*?\]/g
```

Comment:
please move regex outside of the function and make naming in SCREAMING_SNAKE_CASE (only not `get...` :))